Copyright (c) Meta Platforms, Inc. and affiliates.
All rights reserved.

This source code is licensed under the BSD-style license found in the
LICENSE file in the root directory of this source tree.

diff --git a/Python/dynload_shlib.c b/Python/dynload_shlib.c
index c51f97abd2..83f73e351d 100644
--- a/Python/dynload_shlib.c
+++ b/Python/dynload_shlib.c
@@ -54,8 +54,7 @@ static struct {
 } handles[128];
 static int nhandles = 0;

-
-dl_funcptr
+dl_funcptr __attribute__((weak))
 _PyImport_FindSharedFuncptr(const char *prefix,
                             const char *shortname,
                             const char *pathname, FILE *fp)
